test/classifier_sequence_test.py

```diff
@@ -0,0 +1,68 @@
+import collections
+import pytest
+from test import icon_builder as ib
+from src import ClassifierSequence
+
+
+@pytest.fixture
+def input_dir(tmp_path_factory):
+    return tmp_path_factory.mktemp("input_dir")
+
+
+@pytest.fixture
+def icon_builder(input_dir):
+    return ib.IconBuilder(input_dir)
+
+
+class TestClassifierSequence:
+    @pytest.mark.parametrize("num_icons, expected_len", [
+        (6, 2),  # 6 icons, batch size 3 => 2 batches
+        (7, 3),  # 7 icons, batch size 3 => 3 batches
+        (5, 2),  # 5 icons, batch size 3 => 2 batches
+    ])
+    def test_len(self, icon_builder, num_icons, expected_len):
+        classifier_sequence = _get_sequence(icon_builder, num_icons=num_icons, batch_size=3)
+
+        assert len(classifier_sequence) == expected_len
+
+    def test_all_batches_have_full_batch_size(self, icon_builder):
+        batches = []
+        batch_size = 2
+        classifier_sequence = _get_sequence(icon_builder, batch_size=batch_size, num_icons=3)
+
+        for batch_x, batch_y in classifier_sequence:
```

Comment:
you could also do

```
batches = [batch for batch in [batch_x, batch_y] for batch_x, batch_y in classifier sequence]
```

or

```
batches = [batch for batch in batches for batches in classifier sequence]
```

Not sure if that is really easier to read.

An alternative would be:

```
for batch_x, batch_y in classifier_sequence:
    assert len(batch_x) == len(batch_y) == batch_size
```

Of all solutions that would be my favorite, since it also removes the a bit convoluted `all` statement in the end.  
The downside of this is that the test breaks early if one batch is wrong and does not give any information about the other batches.